data/core/macros/abilities.cfg

```diff
@@ -257,7 +257,7 @@ Any units adjacent to this unit will fight as if it were dusk when it is night,
         [tunnel]
             id=village_teleport
             [source]
-                terrain=*^V*
+                gives_income=true
```

Comment:
There are no _core_ terrains that give income and don't match `*^V*`, so this seems fine to me, but are there any add-ons that have such terrains?

Reply:
gives_income is the definition of village. ^V is just code style convention.

A hex is considered a village if and only if its [ [terrain_type](https://wiki.wesnoth.org/TerrainWML) ]gives_income= parameter was set to yes (which means a side can own that hex).